File: solid_stress_der.m
function v = solid_stress_der(phi)

global diff_idx;

if (diff_idx == 1)
    v = sige_der(phi);
elseif (diff_idx == 2)
    v = sige_reg_der(phi);
else 
    error('Undefined diff_idx!!');
end

File: solid_stress_der.cpp
#include "solid_stress_der.h"

double solid_stress_der(double phi)
{
    if (global::diff_idx == 1)
        return sige_der(phi);
    else if (global::diff_idx == 2)
        return sige_reg_der(phi);
    else
        throw std::invalid_argument("\nUndefined diff_idx!!\n"); 
}